cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/services/ApplicationLifecycleServiceTest.java

```diff
@@ -452,6 +473,102 @@ public void testCreateAppDetailsArchive() throws Exception {
     deleteNamespace("ns3");
   }
 
+  @Test(expected = ApplicationNotFoundException.class)
+  public void testDeployWithoutMarkingAppAsLatest() throws Exception {
+    String appName = "application_not_marked_latest";
+    Id.Application appId = Id.Application.from(Id.Namespace.DEFAULT, appName);
+    Id.Artifact artifactId = Id.Artifact.from(Id.Namespace.DEFAULT, "appWithConfig", "1.0.0-SNAPSHOT");
+    addAppArtifact(artifactId, ConfigTestApp.class);
+    HttpResponse resp = deployWithoutMarkingLatest(
+        appId, new AppRequest<>(ArtifactSummary.from(artifactId.toArtifactId())));
+    Assert.assertEquals(200, resp.getResponseCode());
+
+    applicationLifecycleService.getLatestAppDetail(appId.toEntityId().getAppReference());
+  }
+
+  @Test
+  public void testMarkAppsAsLatest() throws Exception {
+    setLcmEditFlag(false);
```

Comment:
Why are we setting lcm as false here ? without lcm marklatest has no usecase

Reply:
Updated the test and removed the line setting lcmEditFlag to false.